### DISCUSSION TITLE
VirtualEnv missing in 14?

### DISCUSSION THREAD

**Discussion by brettc** [LazyVim/LazyVim#5081] [Q&A] [answered]
Hi. I just bumped to 14 and my usual  `<leader>cv` does not show virtualenv selection for python any more. Was this dropped?

**Comment by dpetka2001** [comment, marked as answer]
This is only enabled for Telescope. With the recent update `fzf-lua` was made the default file picker and that is not compatible with `venv-selector`. There's an open issue at their github repo about supporting `fzf-lua` that you might want to track. In the meantime you can just use `:LazyExtras` to re-enable Telescope if `venv-selector` is crucial to your workflow.

**Reply by brettc** [reply]
Thanks for the speedy reply!

**Reply by benjaminbauer** [reply]
For the time beeing, this breaks python support in LazyVim for me. The visual selector is nice to have, but the automatic selection of the venv is important for e.g. LSP functionality.

**Reply by brettc** [reply]
Enabling telescope via `:LazyExtras` does the trick. Or you can put `{ "LazyVim", version = "13" },` into your config.

**Reply by dpetka2001** [reply]
If you do `{ "LazyVim", version = "13" }` and leave it like that for a long time and then decide to update, I bet you will have to deal with lots of things. And you also have to consider that LazyVim also adopts some Neovim nightly changes and when Neovim 0.11 stable comes out some things might be broken and you may have to stay in Neovim 0.10. 

Just a couple of things to keep in mind.

**Reply by nicolalicata-upside** [reply]
Are there any alternatives to `venv-selector` that work with `fzf-lua` or am I just stuck with using Telescope for now?

**Reply by dpetka2001** [reply]
To my knowledge (and I could be wrong) no there's not. You would have to wait until this [feature request](https://github.com/linux-cultist/venv-selector.nvim/issues/142) gets implemented.

**Reply by patelharsh9797** [reply]
for the easy fix till the [venv-fzf-feature](https://github.com/linux-cultist/venv-selector.nvim/issues/142) get updated, you can use the telescope as decency and it's working fine with fzf for all other works and just a telescope for venv-selector.

```
{
    "linux-cultist/venv-selector.nvim",
    dependencies = { "nvim-telescope/telescope.nvim" },
    config = function()
      require("venv-selector").setup()
    end,
}
```

**Reply by dpetka2001** [reply]
For those who use `venv-selector`, could you try this plugin I read about on Reddit today [whichpy.nvim](https://github.com/neolooong/whichpy.nvim) and share your thoughts if it's a good alternative for `venv-selector` to use with `fzf-lua`?

**Reply by Jacobus-afk** [reply]
Something like [direnv](https://direnv.net/), that can automatically activate your virtual environment as soon as your cd into your project directory, might also work for you?

**Reply by benjaminbauer** [reply]
I think the best solution would be to not disable venv-selector, but only disable the keybinding that opens the visual selector that currently only works with telescope.

**Reply by dpetka2001** [reply]
@benjaminbauer PR is welcome with your reason to do so and maintainer will decide after he's back from vacation. And users can still put that in their personal configuration if they wish so. It was decided to be like that because users were opening issues that it wouldn't work with fzf-lua even when Telescope was the default picker back then.

**Reply by Flimm** [reply]
> To my knowledge (and I could be wrong) no there's not. You would have to wait until this [feature request](https://github.com/linux-cultist/venv-selector.nvim/issues/142) gets implemented.

It looks like venv-selector implements support for fzf-lua now: https://github.com/linux-cultist/venv-selector.nvim/issues/142 and https://github.com/linux-cultist/venv-selector.nvim/pull/188 You need to use the `regexp` branch rather than the `main` branch.
